File: palindrome/valid_palindrome.py
"""
125. Valid Palindrome
Solved
Easy
Topics
Companies
A phrase is a palindrome if, after converting all uppercase letters into lowercase letters and removing all non-alphanumeric characters, it reads the same forward and backward. Alphanumeric characters include letters and numbers.

Given a string s, return true if it is a palindrome, or false otherwise.

"""

import re
class Solution:
    def isPalindrome(self, s: str) -> bool:
        clean_s = re.sub('[^A-Za-z0-9]', '', s)
        return clean_s.lower() == clean_s[::-1].lower()

File: palindrome/valid_palindrome.go
/*
125. Valid Palindrome

A phrase is a palindrome if, after converting all uppercase letters into lowercase letters and removing all non-alphanumeric characters, it reads the same forward and backward. Alphanumeric characters include letters and numbers.
Given a string s, return true if it is a palindrome, or false otherwise
*/
package palindrome

import (
	"go_algos/sliceutils"
	"regexp"
	"strings"
)

func isPalindrome(s string) bool {
	clean_s := regexp.MustCompile("[^A-Za-z0-9]").ReplaceAllString(s, "")
	//    if len(clean_s) == 0 {return false}
	return strings.ToLower(clean_s) == strings.ToLower(sliceutils.StringReverse(clean_s))
}
